### DISCUSSION TITLE
(Windows) Physical rotated screens output to rustdesk is rotated, mouse movement is rotated too

### DISCUSSION THREAD

**Discussion by AlvaroNieto** [rustdesk/rustdesk#8322] [Q&A] [answered]
### Bug Description

A monitor is physically rotated, and the user rotates the virtual screen so it matches the rotation of the monitor (Such as a dev with a vertical screen for programming). 

If you connect to that device, Rustdesk will take that virtual configuration explicit into account. Resulting in trying to use the screen sideways or upside-down, and with the mouse movement messed up. 

### How to Reproduce

Rotate a physical monitor, on Windows rotate the image so it matches the new rotation of the physical monitor. Connect and notice that the virtual signal is rotated so you have to turn your head to read something, but the mouse movement is (for example, if the screen is rotated 180º, you will have the mouse inverted).

Windows 10, not tested in Windows 11.

### Expected Behavior

Rustdesk should ignore, or correct any image rotation configuration so the remote control is not affected. 

### Operating system(s) on local side and remote side

Windows 10

### RustDesk Version(s) on local side and remote side

1.2.5 -> 1.2.5

### Screenshots

I am upgrading the Windows 10 to 11 to see if the problem persists. I will post the pictures once I have that computer running.

**Comment by AlvaroNieto** [comment]
The screenshot as promised, the issue persists in Windows 11.

![image](https://github.com/rustdesk/rustdesk/assets/16096292/0392184b-d6db-493b-b542-d69e37c154dd)

After further testing. **I can confirm the control works fine. The mouse placement equals the mouse placement in the client.** The only issue is the rotated screen. It should ignore or correct the image to an always "fixed" position.

**Comment by Vertganti** [comment]
We are affected by the same issue on a device with three screens. Two of the screens are physically rotated and the image has been rotated back in Windows (Landscape flipped setting). On the two flipped screens the input is mirrored as described by @AlvaroNieto . The third screen (non-rotated) is working as expected. We have found no setting which allows us to fix this behaviour. As I can see it there are two problems here:

- When a display is rotated the controls are not rotated resulting in mirrored controls.
- There is no way to rotate a displayed Windows client screen in the RustDesk host client.

**Comment by rustdesk** [comment]
@fufesou note this, lower priority than your current issues.

**Reply by rustdesk** [reply]
also https://github.com/rustdesk/rustdesk/issues/9301

**Comment by fufesou** [comment]
@AlvaroNieto @Vertganti 

Only this orientation has this issue?

<img width="162" alt="d73d03f6eb86e20cefb18f9cc2a07b5" src="https://github.com/user-attachments/assets/a48890f0-cd2b-4c2e-855b-a3a7cdaf2b51">

How about the other two "Portait" orientation?

**Reply by fufesou** [reply]
Could you also test if this build (on the **controlled side**)  works fine ?

**Reply by Vertganti** [reply]
I currently don't have a spare device to test the portrait rotation settings. I'll see if I can get my hands on one.

> Could you also test if this build (on the **controlled side**) works fine ?

Which build are you referring to? 1.3.1 or am I missing the link?

**Reply by fufesou** [reply]
Sorry https://github.com/fufesou/rustdesk/releases/tmp

**Reply by Vertganti** [reply]
I have tested the build and it fixed the issue for the tested device. The client side had one normal Lanscape screen and another which was configured to be Landscape (flipped). On the controlling side both were displayed as normal Landscape and mouse positions were correct.

**Reply by AlvaroNieto** [reply]
Not working for me. Tried that version both in controlled and client side. The result is the same.

If it helps, I am connecting from a Vega 11 integrated card to a workstation with RX550 (the one with landscape flipped).

**Reply by fufesou** [reply, marked as answer]
@AlvaroNieto Thanks for your feedback.

Could you please try the latest nightly build?

How about the Portrait orientation?

**Reply by AlvaroNieto** [reply]
Cheers mate. The nightly one has made it work flawlessly. Both portrait and landscape.
